### ISSUE TITLE
Cannot auto_init a repo created from template

### ISSUE THREAD

**Issue by ewan-chalmers** [google/go-github#2291]
Having used `Repositories.CreateFromTemplate`, I cannot then successfully use  `Repositories.Edit`. It gives error
```
422 Unprocessable Entity response
PATCH <repo_url>: 422 Validation Failed [{Resource:Repository Field:default_branch Code:invalid Message:Cannot update default branch for an empty repository. Please init the repository and push first.}]
```

Is this a bug with the github API rather than a limitation of this lib?

**Comment by gmlewis** [comment]
Interesting!  It sounds like the repo needs some content first.  Does your template provide content to the repo?
If so, have you tried adding in a delay before performing the edit?

In the past, I was doing some creation operations using the GitHub v3 API and I remember having to insert a loop (with exponential backoff) with an intervening "Read/Verify" step repeatedly checking for the content I created before being able to edit.

I contacted GitHub tech support at the time and they told me that they had a bunch of replicated front-end API servers that were not always immediately sync'd... it would take some time for the changes to propagate to all the servers ("eventually consistent").  I'm not sure if you are running into the same thing, but you might want to contact GitHub tech support and see what they say.  Feel free to report back here when you find out anything... and good luck.
